src/robot/libraries/dialogs_py.py

```diff
@@ -68,18 +68,18 @@ def _finalize_dialog(self):
         if self.widget:
             self.widget.focus_set()
 
-    def _create_body(self, message, value, **config) -> Union[Entry, Listbox, None]:
+    def _create_body(self, message, value, default: Union[int,str,None]=None, **config) -> Union[Entry, Listbox, None]:
         frame = Frame(self)
         max_width = self.winfo_screenwidth() // 2
         label = Label(frame, text=message, anchor=W, justify=LEFT, wraplength=max_width)
         label.pack(fill=BOTH)
-        widget = self._create_widget(frame, value, **config)
+        widget = self._create_widget(frame, value, default, **config)
```

Comment:
As explained above, you shouldn't need to modify `TkDialog` at all. I'm actually surprised if this didn't break other dialogs, because the `_create_widget` call isn't compatible with subclass methods. Did you try running all acceptance tests with your changes?

Reply:
[aef0c56](https://github.com/robotframework/robotframework/pull/5182/commits/aef0c56dc9b11c5562939bf7cce71072a922b8da)